### ISSUE TITLE
[Bug]: cnv:callCNV chokes on a CRAM made internally by the workflow

### ISSUE THREAD

**Issue by mp15** [epi2me-labs/wf-human-variation#38]
### What happened?

I ran the workflow with the following commandline: 
```
../nextflow -c local.config run epi2me-labs/wf-human-variation     -w /fastdata/152_nextflow_workspace     -profile standard     --snp --cnv --sv     --bam data/OESO_152_sup_4.1.0.pass.map_sort.nomoves.bam      --ref ../resources/GCA_000001405.15_GRCh38_full_analysis_set/GCA_000001405.15_GRCh38_full_analysis_set.fa      --sample_name WTSI-OESO_152     --out_dir /fastdata/OESO_152_wf-human-variation
```
The CNV workflow died with a complaint about CRAM files. You'll notice I'm not feeding it a CRAM in the first place, I'm feeding it a BAM, it made the CRAM itself internally to the workflow! Does that mean the CNV component just doesn't work in this version of wf-human-variation?

This may be related to #25 however I did not feed it the CRAM in the first place so it's definitely a bug rather than an enhancement.

### Operating System

ubuntu 20.04

### Workflow Execution

Command line

### Workflow Execution - CLI Execution Profile

Docker

### Workflow Version

wf-human-variation v1.4.0-gd1eb43d

### Relevant log output

```shell
Error executing process > 'cnv:callCNV (1)'

Caused by:
  Process `cnv:callCNV (1)` terminated with an error exit status (1)

Command executed:

  run_qdnaseq.r --bam WTSI-OESO_152.cram --out_prefix WTSI-OESO_152 --binsize 500 --reference hg38
  cut -f5 WTSI-OESO_152_calls.bed | paste WTSI-OESO_152_bins.bed - > WTSI-OESO_152_combined.bed
  check=`awk -F "       " 'NF != 6' WTSI-OESO_152_segs.seg`

  if [ -n "${check}" ]; then             echo "vcf is malformed";             workflow-glue fix_vcf --vcf WTSI-OESO_152_calls.vcf --fixed_vcf WTSI-OESO_152_calls_fixed.vcf --sample_id WTSI-OESO_152;             workflow-glue fix_vcf --vcf WTSI-OESO_152_segs.vcf --fixed_vcf WTSI-OESO_152_segs_fixed.vcf --sample_id WTSI-OESO_152;
      rm WTSI-OESO_152_calls.vcf
      mv WTSI-OESO_152_calls_fixed.vcf WTSI-OESO_152_calls.vcf
      rm WTSI-OESO_152_segs.vcf
      mv WTSI-OESO_152_segs_fixed.vcf WTSI-OESO_152_segs.vcf
  fi

Command exit status:
  1

Command output:
  (empty)

Command error:
  Loaded bin annotations for genome 'hg38', bin size 500 kbp, and experiment type 'SR50' from annotation package QDNAseq.hg38 v1.0.0
      WTSI-OESO_152.cram (1 of 1): extracting reads ...Error in value[[3L]](cond) :
    failed to open BamFile: 'filename' is not a BAM file
    file: /fastdata/152_nextflow_workspace/cc/eb60af9ad027f7f31e824b6a653e82/WTSI-OESO_152.cram
  Calls: binReadCounts ... tryCatch -> tryCatchList -> tryCatchOne -> <Anonymous>
  Execution halted

Work dir:
  /fastdata/152_nextflow_workspace/7e/89fc35d7c2f3c4843139aeb0f2154b

Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh`
```

**Comment by SamStudio8** [comment]
#25 is rather generously categorised as an enhancement issue but likewise we are working to rectify the incompatibility with the CNV subworkflow. Did your workflow instance perform (re)alignment? Only the basecalling and alignment components generate CRAM, an aligned BAM with the correct reference should otherwise have been passed through.

**Comment by mp15** [comment]
The input BAM `OESO_152_sup_4.1.0.pass.map_sort.nomoves.bam` is a mapped bam and wasn't supposed to trigger the remapping workflow but I checked and it did trigger minimap2 yes. Any idea why?

**Comment by SamStudio8** [comment]
The workflow will trigger remapping if it thinks that the `--ref` did not match the alignments in the input BAM. There should be a warning in the stdout and your Nextflow log if this was encountered. You could also grep the Nextflow log to get the workdir for the `check_for_alignment` process and find the logs in that directory.

**Comment by ymcki** [comment]
I also have this problem but I started from the HG002 pod5 raw data. Please fix. Or maybe I should just remove -O CRAM in basecalling.nf to force it to output bam???

**Comment by SamStudio8** [comment]
Thanks both, this is a limitation in the downstream software used in the CNV subworkflow. We have just released wf-human-variation 1.6.0 which addresses this in the short term by internally converting CRAM to BAM when calling for CNVs. Please let me know how you get on with this.
